master/internal/db/postgres_groups.go

```diff
@@ -30,22 +35,28 @@ func (db *PgDB) SearchGroups(ctx context.Context, userBelongsTo model.UserID) ([
 	}
 
 	err := query.Scan(ctx)
+
+	if len(groups) < 1 || errors.Is(err, sql.ErrNoRows) {
```

Comment:
This is a judgment call, but I think we shouldn't return `ErrNotFound` when we search for groups or users but find nothing. In my opinion, it's not necessarily an error condition when we look for a set of things and the set is empty. By analogy, if I ran a command like this:

```
find / -name 'some-weird-file-name.go'
```

and had no file in my filesystem with that name, I'd expect the `find` command to exit with code 0 rather than 1; it successfully ran its search, and returned correct results: that the set of matches was empty.

Reply:
yeah that makes sense - I changed this to return whatever the query returned